Feature/basic setup

fedgut opened the pull request:
Requesting a code review to merge to development

fedgut committed:
Create basic user and log in

men32z committed:
helpers and controles for login

fedgut committed:
fix basic log in

fedgut committed:
Create post model

fedgut committed:
Create user and post asossiations

fedgut committed:
Update README.md

fedgut committed:
Correct stickler errors

fedgut committed:
fix additional linting mistake

fedgut committed:
Remove tailing whitspace

adriaanbd commented on config/routes.rb:
```diff
@@ -1,3 +1,12 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
```
As specified in the Additional Microverse Requirements section of [this project](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) in our Curriculum, please add a Sign-Up feature in order to be able to create new users from the UI:

![image](https://user-images.githubusercontent.com/22228397/69654831-7204ec80-1043-11ea-8f0f-b5e6a1c0c3de.png)

This is [`REQUIRED`]

adriaanbd commented on db/migrate/20191125160749_create_users.rb:
```diff
@@ -0,0 +1,15 @@
+    create_table :users do |t|
+      t.string :email
+      t.string :username
+      t.boolean :activated
+      t.boolean :admin
```
I'm curious about these two attributes:
* Did you add them?
* Are you using them?
* Are they required?

men32z replied:
yes, no and  no. we are going to delete them.

adriaanbd commented on db/schema.rb:
```diff
@@ -0,0 +1,34 @@
+    t.string "email"
+    t.string "username"
+    t.boolean "activated"
+    t.boolean "admin"
```
Please refer to the comment above about the use and source of these  two attributes.

adriaanbd commented on app/views/posts/index.html.erb:
```diff
@@ -0,0 +1,11 @@
+<%= e.title %> 
+<%= e.id %>
+  <% if logged_in? %>
+  <%= e.user.username %>
```
If we create a user with `email` and `password` attributes only (without `username`), then this would be `nil` and nothing would be displayed herein. Please note that requirements state only `email` and `password` as requirements to login, but does not mention anything about what fields are required for signing up. However, if you're displaying `username` as author's name of a post, then please make sure that:
1. There's a Sign Up feature
2. That Sign Up requires `username` field

At the moment, Post author is `nil` and nothing is displayed herein; therefore, I can't give you a pass in this requirement because it isn't reflected in the UI, Sign Up form or Model validations, and there's no indication that this is a required attribute beyond being exclusively specified in here and in your `schema`.

As a signed-in user, I shall see the author of a Post [`REQUIRED`]

fedgut replied:
I'm sorry, we didn't quite got this one! Could you help us to clarify if we fixed it properly?

adriaanbd commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,21 @@
+    redirect_to(session[:forwarding_url] || user)
+  end
+
+  def sing_out
```
Sing Out? Really? I mean, if you want to name it like that, by all means, go ahead, but I initially thought this was a typo or a mistake on your part; thanks for the :rofl:

adriaanbd commented on app/views/sessions/new.html.erb:
```diff
@@ -0,0 +1,11 @@
+<%= form_with  local: true do |f| %>
```
Requirements state that this form needs to be created with a `form_for` [`REQUIRED`].

men32z replied:
that was deprecated, read the following documentation. https://guides.rubyonrails.org/form_helpers.html#using-form-for-and-form-tag

fedgut replied:
Hello!  Rails documentation says that form_for is deprecated! Should we revert back to the deprecated form_for?

fedgut replied:
"form_for. form_tag and form_for are soft deprecated and they will be replaced by form_with in the future."

adriaanbd replied:
If it is soft deprecated then our guidelines need to be updated. Keep the `form_with` and don't change it.

adriaanbd reviewed (changes requested):
A few changes are required to be able to move forward:
* Please add a Sign Up feature
* New Post uses `form_for`
* Signed in user sees Post's author

Please refer to the comments below for more information.

About GitFlow: 

* The `development` branch should be set up as your *default* branch.
* The *master* branch should keep the release version of your product after the feature branches have been merged with the development branch (at the moment, your master branch has a bunch of files and folders from your `rails` project setup. Since those files are not final, and the setup could change, depending on your first release, it is better to commit them afterward). At most, you'd want to have only a README and your linters config files for them to work properly. Please keep this in mind for your next project.

Status: `Changes Requested`
Reviewer: `Adriaan Beiertz`

Feel free to contact me via Slack, email or initiate discussion herein.

men32z committed:
users unused fields deleted

men32z committed:
following changes
the username field is required and validated.
there is an interface to create users.

adriaanbd reviewed (approved):
Thanks for incorporating the previously suggested changes.

Status: `Approved`
Reviewer: `Adriaan Beiertz`
